src/Abstractions/NexusMods.Abstractions.NexusModsLibrary/Models/CollectionRevision.cs

```diff
@@ -0,0 +1,55 @@
+using NexusMods.Abstractions.MnemonicDB.Attributes;
+using NexusMods.Abstractions.NexusModsLibrary.Attributes;
+using NexusMods.MnemonicDB.Abstractions.Attributes;
+using NexusMods.MnemonicDB.Abstractions.Models;
+
+namespace NexusMods.Abstractions.NexusModsLibrary.Models;
+
+/// <summary>
+/// Metadata about a collection revision on Nexus Mods. A revision references a collection, but is itself immutable.
+/// Each change to a collection is expressed as a separate revision.
+/// </summary>
+public partial class CollectionRevision : IModelDefinition
+{
+    private const string Namespace = "NexusMods.Library.NexusModsCollectionRevision";
+    
+    /// <summary>
+    /// The globally unique id identifying a specific revision of a collection.
+    /// </summary>
+    public static readonly RevisionIdAttribute RevisionId = new(Namespace, nameof(RevisionId)) { IsIndexed = true };
+    
+    /// <summary>
+    /// The locally unique revision number (aka "version") of a collection. Only unique within one collection.
+    /// </summary>
+    public static readonly RevisionNumberAttribute RevisionNumber = new(Namespace, nameof(RevisionNumber));
+    
+    /// <summary>
+    /// The collection this revision belongs to.
+    /// </summary>
+    public static readonly ReferenceAttribute<Collection> Collection = new(Namespace, nameof(Collection));
+    
+    /// <summary>
+    /// The number of downloads this revision has.
+    /// </summary>
+    public static readonly ULongAttribute Downloads = new(Namespace, nameof(Downloads));
+    
+    /// <summary>
+    /// Total size of all files in this revision, including the size of the revision's files.
+    /// </summary>
+    public static readonly SizeAttribute TotalSize = new(Namespace, nameof(TotalSize));
```

Comment:
I'm still not sure what size this refers to. Is this "download size" or "install size"? Wabbajack has both since we index and hash everything, I can imagine that download size would work for collections, but I can't see install size being anything more than an estimate.

Reply:
Good point, added clerification